### ISSUE TITLE
Project Settings Not stored

### ISSUE THREAD

**Issue by magdesign** [NatronGitHub/Natron#439]
Natron 2.3.15 Snapshot (RC11 and RC12)
Ubuntu 18.04 LTS (PopOS)

Everytime I open a project in Natron, the project automatically is set to:
```
Output Format  1920x1080 pixel
25fps
```
Even if the project was stored with 
```
Output Format  1920x1200
30fps
```
It seems that saving a file does not save project settings at all, especially resolution and frame rate.

**Comment by rodlie** [comment]
Project settings are saved, example UHD_4K 30:
```
<item>
		<Name>outputFormat</Name>
		<Type>Choice</Type>
		<Dimension>1</Dimension>
		<Secret>0</Secret>
		<MasterIsAlias>0</MasterIsAlias>
		<item>
			<Enabled>1</Enabled>
			<HasAnimation>0</HasAnimation>
			<Value>7</Value>
			<Default>6</Default>
			<HasMaster>0</HasMaster>
			<Expression></Expression>
			<ExprHasRet>0</ExprHasRet>
		</item>
		<ChoiceLabel>UHD_4K</ChoiceLabel>
		<UserKnob>0</UserKnob>
	</item>
	<item>
		<Name>frameRate</Name>
		<Type>Double</Type>
		<Dimension>1</Dimension>
		<Secret>0</Secret>
		<MasterIsAlias>0</MasterIsAlias>
		<item>
			<Enabled>1</Enabled>
			<HasAnimation>0</HasAnimation>
			<Value>3.00000000000000000e+01</Value>
			<Default>2.40000000000000000e+01</Default>
			<HasMaster>0</HasMaster>
			<Expression></Expression>
			<ExprHasRet>0</ExprHasRet>
		</item>
		<UserKnob>0</UserKnob>
	</item>
```

Project settings are restored, but the combo box is not set correctly.

![project-restore](https://user-images.githubusercontent.com/34516798/70811631-46fdf500-1dc6-11ea-8d9d-5287ec1e453d.png)

**Comment by magdesign** [comment]
It does not really save them, you recognize it when there is a ramp and/or a reformat node in then scene. 
I am working on 1920x1200, everytime I open the scene, the ramp and all reformats are on 1920x1080 as well as the scene settings, which results in renders with black borders.

**Comment by rodlie** [comment]
Are the ramp/reformat ok after you change the default format in the project settings to 1920x1200?

**Comment by magdesign** [comment]
Yes, they are.

**Comment by rodlie** [comment]
Then my patch should work, I will double check a bit later today (I'm not in front of my main dev machine at the moment).

**Comment by splidje** [comment]
Hello. I have a branch here called #439, which I believe is a fix:
https://github.com/splidje/Natron

**Comment by magdesign** [comment]
Is it possible to get a linux binary?

**Comment by devernay** [comment]
@rodlie do you have a patch for this issue? Should we consider this high-priority?

**Comment by rodlie** [comment]
Yes, but it seems the issue was not resolved for @magdesign. I'm currently not able to replicate the issue he has, so please (@magdesign) send me a simplified project where I can replicate your issue.

**Comment by magdesign** [comment]
Here the link: https://cloud.magdesign.ch/index.php/s/ztRntC6wxZsmtZj
Project is saved as 1920x1200
Frame range: 1 last 1500
Frame Rate: 30

**Comment by rodlie** [comment]
Thanks, will review issue later tonight when I get home.

**Comment by magdesign** [comment]
Edit: The thing with the Reformat I just realized that in the other projects I did not select "Distort" in the dropdown, but framerate and scene range issue as well as scene settings remain on HD persist. Thanks for looking in to it, very appreciated!

**Comment by rodlie** [comment]
The patch from @splidje (https://github.com/splidje/Natron/commit/2cabb20bd230a4f02141986a57875737493d1c6c) works here.

**Comment by devernay** [comment]
Great! Let us pull it then.
@splidje please tell us if you want to appear in the list of contributors, and under which name.

**Comment by splidje** [comment]
@devernay, yes please. Could you put "Jonathan Hearn"

**Comment by rodlie** [comment]
Issue is now considered fixed, close.

**Comment by devernay** [comment]
Thank you @splidje added you in https://github.com/NatronGitHub/Natron/commit/e8d25fb3b3248bb9953d4edd22c95a02511c9a15

**Comment by magdesign** [comment]
Just built the newest version e8d25fb (Git branch RB-2.3" ) and tested.
The issue persists, is the fix from plidje in another branch?

See video demonstration of the issue here: https://vimeo.com/383488301
password: natron

Creating a comp witt settings: 1920x1200, 1500frames long, 30fps
saving the comp, close Natron, open saved comp again, scene settings lost.

Ubuntu 18.04LTS, PopOS

**Comment by rodlie** [comment]
Well, that's odd. It works fine here with the patch applied.
[simplescreenrecorder-2020-01-08_16.58.14.mkv.zip](https://github.com/NatronGitHub/Natron/files/4036225/simplescreenrecorder-2020-01-08_16.58.14.mkv.zip)

**Comment by magdesign** [comment]
Maybe I did something wrong with building.... any change to get a compiled version?

**Comment by devernay** [comment]
@magdesign @rodlie be careful, I suspect the fix caused a regression: https://github.com/NatronGitHub/Natron/issues/447

**Comment by devernay** [comment]
@splidje do you have an idea why it may have broken some projects? (see #447)

**Comment by devernay** [comment]
confirmed: this fix caused #447 
I propose to revert this fix, and close #447 
We have to find a solution that does not break loading existing projects.

The following unit tests are broken by this fix:

TestFrameBlend : FAIL
TestMultiPlaneEXR : FAIL
TestMultiPlaneORA : FAIL
TestMultiPlanePSD : FAIL
TestMultiPlaneXCF : FAIL
TestRetimeTransform : FAIL
TestTimeBlur : FAIL
TestZMask : FAIL

**Comment by devernay** [comment]
https://github.com/NatronGitHub/Natron/commit/838ebf1b3444f6b70b84d1606cddbdd945539b49 reverted changed from 2cabb20bd230a4f02141986a57875737493d1c6c

**Comment by devernay** [comment]
please run unit tests before committing a new fix, and thanks to everyone who can help on this.
If you don't know how to run unit tests, simply load the project mentionned in https://github.com/NatronGitHub/Natron/issues/447 and check if the output channels look ok in the nodegraph

**Comment by rodlie** [comment]
I'm sorry for not running the unit tests, will do so in the future. I have time tomorrow to help look more into this issue.

**Comment by devernay** [comment]
that's ok @rodlie ! we're lucky that I took a look at the test results. I'll also more consistently post the test results with the binaries, since they can be very helpful.

**Comment by devernay** [comment]
This should be fixed by #450, @rodlie @splidje @magdesign can you please test that the latest RC (RC15) fixes this?

**Comment by magdesign** [comment]
Scene settings are loading correct with RC15 (on Ubuntu 18.04 Linux)
Thank you!

frameblend.ntp renders without errors, 
but do not exactly know how to check what was wrong with the frameblending node,
looks like this: https://user-images.githubusercontent.com/3778071/72196802-9d635100-33cf-11ea-8e14-ccd992127aca.png
